mapper/orm/src/main/java/org/hibernate/search/mapper/orm/event/impl/HibernateSearchEventListener.java

```diff
@@ -139,16 +144,15 @@ public void onPostUpdateCollection(PostCollectionUpdateEvent event) {
 	public void onFlush(FlushEvent event) {
 		getCurrentWorkPlan( state.getContextProvider(), event.getSession() ).prepare();
 
-		// TODO HSEARCH-3068 handle the "simulated" transaction when a Flush listener is registered
-//		Session session = event.getSession();
-//		Synchronization synchronization = flushSynch.get( session );
-//		if ( synchronization != null ) {
-//			//first cleanup
-//			flushSynch.remove( session );
-//			log.debug( "flush event causing index update out of transaction" );
-//			synchronization.beforeCompletion();
-//			synchronization.afterCompletion( Status.STATUS_COMMITTED );
-//		}
+		Session session = event.getSession();
+		Synchronization synchronization = flushSynch.get( session );
```

Comment:
I don't know if it a good idea always calling execute on the work plan. In such way we could invoke execute on flush even if there is no update outside transactions on the current session.

It would nice to have if there was a way to detect if there are some out of transaction update operations pending on the current session.

Currently we rely on the weird synchronization map to detect them. If there was another way, I would happy about it.